phpdocs/files/Config.php.txt
<?php
/**
 * App configuration
 *
 * @author  Jared Howland <contacts@jaredhowland.com>
 * @version 2017-12-11
 * @since   2016-09-28
 *
 */

namespace Contacts;

/**
 * Configuration class to import `Config.ini` file and set other defaults
 */
class Config
{
    /**
     * @var array $config Array of values in `Config.ini`
     */
    private static $config;

    /**
     * Set error reporting based on if app is in development or production
     *
     * @param null
     *
     * @return void
     */
    public static function setErrorReporting()
    {
        if (self::get('development')) {
            ini_set('display_errors', '1');
            ini_set('error_reporting', E_ALL ^ E_NOTICE);
        } else {
            ini_set('display_errors', '0');
        }
    }

    /**
     * Get config setting from `.ini` file
     *
     * @param string $setting Setting name
     *
     * @return string Return setting value from `.ini` file
     */
    public static function get(string $setting): string
    {
        if (empty(self::$config)) {
            self::$config = parse_ini_file('Config.ini');
        }

        return self::settingExists($setting);
    }

    /**
     * Check if setting in `.ini` file exists
     *
     * @param string $setting Setting name to check for
     *
     * @throws \UnexpectedValueException when `$setting` does not exist
     *
     * @return string Setting value from `.ini` file
     */
    private static function settingExists(string $setting): string
    {
        if (isset(self::$config[$setting])) {
            return self::$config[$setting];
        } else {
            throw new \UnexpectedValueException("'$setting' is not a valid config setting. Please check your 'config.ini' file for valid config options.\n");
        }
    }
}

config::setErrorReporting();



Documentation/files/Config.php.txt
<?php
/**
 * App configuration
 *
 * @author  Jared Howland <contacts@jaredhowland.com>
 * @version 2017-12-11
 * @since   2016-09-28
 *
 */

namespace Contacts;

/**
 * Configuration class to import `Config.ini` file and set other defaults
 */
class Config
{
    /**
     * @var string $config String returned by `Config.ini`
     */
    private static $config;

    /**
     * Set error reporting based on if app is in development or production
     *
     * @param null
     *
     * @return void
     */
    public static function setErrorReporting()
    {
        if (self::get('development')) {
            ini_set('display_errors', '1');
            ini_set('error_reporting', E_ALL ^ E_NOTICE);
        } else {
            ini_set('display_errors', '0');
        }
    }

    /**
     * Get config setting from `.ini` file
     *
     * @param string $setting Setting name
     *
     * @return string Return setting value from `.ini` file
     */
    public static function get(string $setting): string
    {
        if (empty(self::$config)) {
            self::$config = parse_ini_file('Config.ini');
        }

        return self::settingExists($setting);
    }

    /**
     * Check if setting in `.ini` file exists
     *
     * @param string $setting Setting name to check for
     *
     * @throws \UnexpectedValueException when `$setting` does not exist
     *
     * @return string Setting value from `.ini` file
     */
    private static function settingExists(string $setting): string
    {
        if (isset(self::$config[$setting])) {
            return self::$config[$setting];
        } else {
            throw new \UnexpectedValueException("'$setting' is not a valid config setting. Please check your 'config.ini' file for valid config options.\n");
        }
    }

}

config::setErrorReporting();
?>



CHANGELOG.md
Changelog
=========
All notable changes (begining with v3.0.3) to `contacts` will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

[UNRELEASED]
------------

**Added**

- `Read the Docs <https://readthedocs.org>`_ documentation

### Changed ###

- Daisy-chaining methods allowed
- Tests refactored

### Deprecated

- Nothing

### Removed

- Nothing

### Fixed

- Nothing

### Security

- Nothing

## [Pre-v3.0.3]

### Added

- All the things
- Ability to change the directory the `.vcf` file is saved in, the default time zone, and the default area code (for phone numbers missing an area code) when object is created
- Ability to customize the revision date of the `.vcf` file
- Ability to add photos that are URL-referenced or Base64 encoded (all photos are converted to a Base64 encoding to ensure the photo stays with the contact) 
- Ability to let `contacts` generate an unique ID or to pass your own unique ID for a contact
- iOS and macOS-specific vCard fields. These should theoretically work with any other program that supports the full vCard standard but are not guaranteed to operate in the expected manner on those platforms:

  - Anniversary
  - Spouse
  - Child
  - Supervisor
- CHANGELOG.md that follows [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) principles
- CODE_OF_CONDUCT.md from [Contributor Covenant](http://contributor-covenant.org) v1.4 available at <http://contributor-covenant.org/version/1/4/>
- Github templates:

  - CONTRIBUTING.md that provides guidelines on how to contribute to this project
  - ISSUE_TEMPLATE.md for assisting anyone submitting an issue report
  - PULL_REQUEST_TEMPLATE.md that provides a checklist for how to submit a pull request
- Documentation in the `Documentation` directory using [phpDocumentor](https://www.phpdoc.org)
- Example usage in the `Examples` directory
- Unit tests in the `Test` directory
- `.gitattributes` file to slim-down `composer` installations
- `.styleci.yml` to use [StyleCI](https://styleci.readme.io) to enforce [PSR-2 coding style](http://www.php-fig.org/psr/psr-2/)
- `.travis.yml` to automate tests to make sure builds pass all unit tests

### Changed
- `ContactsException` thrown for invalid input instead of failing silently and falling back to default values

### Deprecated
- Method parameters, such as address types, that could be called with either a delimited string or array, are required to be passed as an array now

### Removed
- Nothing

### Fixed
- Code not adhering to PSR-2 coding standards
- Bugs discovered during testing:

  - Time zone offsets that were not correctly validated
  - Geographic coordinates that were not correctly validated

### Security
- Nothing

src/Contacts/Config.php
<?php
/**
 * App configuration
 *
 * @author  Jared Howland <contacts@jaredhowland.com>
 * @version 2017-12-11
 * @since   2016-09-28
 *
 */

namespace Contacts;

/**
 * Configuration class to import `Config.ini` file and set other defaults
 */
class Config
{
    /**
     * @var array|false $config Array of values in `Config.ini`. `false` if file cannot be parsed.
     */
    private static $config;

    /**
     * Set error reporting based on if app is in development or production
     *
     * @param null
     *
     * @return void
     */
    public static function setErrorReporting()
    {
        if (self::get('development')) {
            ini_set('display_errors', '1');
            ini_set('error_reporting', E_ALL ^ E_NOTICE);
        } else {
            ini_set('display_errors', '0');
        }
    }

    /**
     * Get config setting from `.ini` file
     *
     * @param string $setting Setting name
     *
     * @return string Return setting value from `.ini` file
     */
    public static function get(string $setting)
    {
        if (empty(self::$config) && is_array(parse_ini_file('Config.ini'))) {
            self::$config = parse_ini_file('Config.ini');
        }

        return self::settingExists($setting);
    }

    /**
     * Check if setting in `.ini` file exists
     *
     * @param string $setting Setting name to check for
     *
     * @throws \UnexpectedValueException when `$setting` does not exist
     *
     * @return string Setting value from `.ini` file
     */
    private static function settingExists(string $setting)
    {
        if (isset(self::$config[$setting])) {
            return self::$config[$setting];
        } else {
            throw new \UnexpectedValueException("'$setting' is not a valid config setting. Please check your 'config.ini' file for valid config options.\n");
        }
    }
}

config::setErrorReporting();
